src/Microsoft.OData.Core/Uri/ApplyClauseToStringBuilder.cs

```diff
@@ -0,0 +1,170 @@
+                    return ExpressionConstants.KeywordCountDistinct;
+                case AggregationMethod.Max:
+                    return ExpressionConstants.KeywordMax;
+                case AggregationMethod.Min:
+                    return ExpressionConstants.KeywordMin;
+                case AggregationMethod.Sum:
+                    return ExpressionConstants.KeywordSum;
+                case AggregationMethod.VirtualPropertyCount:
+                    return ExpressionConstants.QueryOptionCount;
+                case AggregationMethod.Custom:
+                    return aggExpression.MethodDefinition.MethodLabel;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(aggExpression), "unknown AggregationMethod " + aggExpression.Method.ToString());
+            }
+        }
+
+        private void Translate(AggregateTransformationNode transformation)
+        {
+            bool appendComma = false;
+            foreach (AggregateExpression aggExpression in transformation.Expressions)
+            {
+                appendComma = AppendComma(appendComma);
+
+                AppendExpression(aggExpression.Expression);
+                _query.Append(ExpressionConstants.SymbolEscapedSpace);
+                AppendWord(ExpressionConstants.KeywordWith);
+                AppendWord(GetAggregationMethodName(aggExpression));
+                AppendWord(ExpressionConstants.KeywordAs);
+                _query.Append(aggExpression.Alias);
+            }
+        }
+
+        private void Translate(FilterTransformationNode transformation)
+        {
+            AppendExpression(transformation.FilterClause.Expression);
+        }
+
+        private void Translate(GroupByTransformationNode transformation)
+        {
+            bool appendComma = false;
+            foreach (GroupByPropertyNode node in transformation.GroupingProperties)
+            {
+                if (appendComma)
+                    AppendComma(appendComma);
+                else
+                {
+                    appendComma = true;
+                    _query.Append(ExpressionConstants.SymbolOpenParen);
+                }
+
+                if (node.Expression != null)
+                    AppendExpression(node.Expression);
+
+                bool appendSlash = false;
+                foreach (GroupByPropertyNode childNode in node.ChildTransformations)
+                {
+                    appendSlash = AppendSlash(appendSlash);
+                    AppendExpression(childNode.Expression);
+                }
+            }
+            if (appendComma)
+                _query.Append(ExpressionConstants.SymbolClosedParen);
+
+            if (transformation.ChildTransformations != null)
+            {
+                AppendComma(true);
+                Translate(transformation.ChildTransformations);
+            }
+        }
+
+        public string TranslateApplyClause(ApplyClause applyClause)

```

Comment:
I think this will trigger stylecop error: public method needs to be placed before other private methods.